java/src/com.zeroc.icediscovery/src/main/java/com/zeroc/IceDiscovery/LookupI.java

```diff
@@ -221,7 +221,7 @@ public LookupI(LocatorRegistryI registry, LookupPrx lookup, Properties propertie
         Endpoint[] single = new Endpoint[1];
         for (Endpoint endpt : lookup.ice_getEndpoints()) {
             single[0] = endpt;
```

Comment:
Lots of unnecessary casts leftover from before we changed the `ice_xxx` methods to return a properly downcasted type.